dfastmi/batch/FileNameRetriever.py

```diff
@@ -29,119 +29,195 @@
+        
+    needs_tide : bool
+        Specifies whether the tidal boundary is needed.
 
     Returns
     -------
-    filenames : Dict[Any, Tuple[str,str]]
-        Dictionary of string tuples representing the D-Flow FM file names for
-        each reference/with measure pair. The keys of the dictionary vary. They
-        can be the discharge index, discharge value or a tuple of forcing
-        conditions, such as a Discharge and Tide forcing tuple.
+    filenames : IFileNameRetriever
+        returns the IFileNameRetriever which should be used.
     """
-    filenames: Dict[Any, Tuple[str,str]]
-    key: Union[Tuple[float, int], float]
-    filenames = {}
-    i = 0
-    while True:
-        i = i + 1
-        CSTR = "C{}".format(i)
-        if CSTR in config:
-            Q = float(cfg_get(config, CSTR, "Discharge"))
-            reference = cfg_get(config, CSTR, "Reference")
-            measure = cfg_get(config, CSTR, "WithMeasure")
-            if needs_tide:
-               T = cfg_get(config, CSTR, "TideBC")
-               key = (Q,T)
-            else:
-               key = Q
-            filenames[key] = (reference, measure)
-        else:
-            break
-
-    return filenames
-
-
-def cfg_get(config: configparser.ConfigParser, chap: str, key: str) -> str:
+    if imode == 0 or config is None:
+        return FileNameRetrieverUnsupported()
+    
+    if version.parse(config["General"]["Version"]) == version.parse("1"):
+        return FileNameRetrieverV1()
+    
+    return FileNameRetrieverV2(need_tides)
+
+class FileNameRetrieverUnsupported(IFileNameRetriever):
     """
-    Get a single entry from the analysis configuration structure.
-    Raise clear exception message when it fails.
-
-    Arguments
-    ---------
-    config : Optional[configparser.ConfigParser]
-        The variable containing the configuration (may be None for imode = 0).
-    chap : str
-        The name of the chapter in which to search for the key.
-    key : str
-        The name of the key for which to return the value.
-
-    Raises
-    ------
-    Exception
-        If the key in the chapter doesn't exist.
-
-    Returns
-    -------
-    value : str
-        The value specified for the key in the chapter.
+    File name retriever for unsupported file name retrieving, e.g. for imode = 0.
+    """
+    
+    def get_file_names(self, config : Optional[configparser.ConfigParser] = None) -> Dict[Any, Tuple[str,str]]:
+        """
+        Get file names for unsupported file name retriever.
+
+        Arguments
+        ---------
+        config : Optional[configparser.ConfigParser]
+            The variable containing the configuration (may be None for imode = 0).
+
+        Returns
+        -------
+        filenames : Dict[Any, Tuple[str,str]]
+            Returns and empty Dict.
+        """
+        return {}
+        
+class FileNameRetrieverV1(IFileNameRetriever):
```

Comment:
Is V1 the legacy class and V2 the current implementation? I suggest to use FileNameRetrieverLegacy and FileNameRetriever

Reply:
todo

Reply:
done